static/js/localStorage.js

```diff
@@ -13,20 +13,26 @@ const download = (filename, text) => {
 
 const savingData = () => {
   let inputs = [
-    { name: SolenoidParameters.VOLTAGE, value: '' },
-    { name: SolenoidParameters.LENGTH, value: '' },
-    { name: SolenoidParameters.R0, value: '' },
-    { name: SolenoidParameters.RA, value: '' },
-    { name: SolenoidParameters.X, value: '' },
-    { name: SolenoidParameters.FORCE, value: '' },
-    { name: SolenoidParameters.AWG, value: '' },
+    { name: SolenoidParameters.VOLTAGE, value: '', unit:''},
+    { name: SolenoidParameters.LENGTH, value: '',unit:''},
+    { name: SolenoidParameters.R0, value: '',unit:''},
+    { name: SolenoidParameters.RA, value: '', unit:''},
+    { name: SolenoidParameters.X, value: '', unit:''},
+    { name: SolenoidParameters.FORCE, value: '',unit:''},
+    { name: SolenoidParameters.AWG, value: '',unit:'' },
+    { name: SolenoidParameters.PERMEABILITY, value: '',unit:'' },
   ];
 
   inputs.forEach((input, _index) => {
     let element = document.getElementById(`input-text-${input.name}`);
+    let unit = document.getElementById(`input-unit-${input.name}`);
     if (element.value === '') {
       input.value = 0;
     } else input.value = element.value;
+
+    if(input.name !== SolenoidParameters.VOLTAGE && input.name !== SolenoidParameters.AWG && input.name !== SolenoidParameters.PERMEABILITY) {
+        input.unit = unit.value
+    }
```

Comment:
I think a cleaner way to perform this check would just be to see if unit != null. getElementById() returns null of the WebElement is not found in the document. This would also make it so this conditional does not need to be updated upon future changes

Reply:
+1